### PR TITLE
Added DGContext class

### PR DESCRIPTION
**Pull request by monkeez**
First pass of the DGContext context manager. Let me know if you have any suggestions.

### CODE REVIEW

File: cmdx.py [hunk 1]
```diff
@@ -3917,6 +3914,37 @@ def parent(self, node, parent=None):
...
+
+        if time is not None:
+            if isinstance(time, (int, float)):
+                time = om.MTime(time, om.MTime.uiUnit())
+            super(DGContext, self).__init__(time)
+        else:
+            super(DGContext, self).__init__()
+        self._previousContext = None
+
+    if __maya_version__ >= 2018:
```
Comment (mottosso) [inline cmdx.py:3938]: Hm, will this not break any call to..

```python
with cmdx.DGContext(50):
   # ...
```

If the same code is run on Maya 2017?

An alternative might be to keep the `__enter__` and `__exit__` but to call `cmds.warning("Sorry dude, this no work")` or something like that. Or is there `cmds.error` too? Something loud! :D

**Comment by mottosso** [comment]
I like it.

My initial reaction was that `cmds.Context` would have been cleaner.. and maybe it's worth adding a `Context = DGContext` as another alias below, since there is no other jargon for context than a DG context so seems a bit wasteful. But it would be consistent if you're already familiar with it, like `cmdx.Vector` and `cmdx.TransformationMatrix` etc. So we shoul definitely keep it in.

File: cmdx.py (hunk 1 above)
Comment (monkeez) [reply cmdx.py]: Yeah that's correct. I've just pushed an update, is that what you had in mind?

Comment (mottosso) [reply cmdx.py]: Yes, that's it. Thanks

**Comment by mottosso** [comment]
I think this looks good, merging as soon as the (needless) tests passes (for the README badges to not break).

**Pull request merged**